lambda-packages/dns-validated-domain-identity-handler/src/verifier.ts

```diff
@@ -70,10 +86,24 @@ export class Verifier {
       Identity: this.domainName,
     }).promise();
 
-    console.log("Deleting DNS Records used for domain verification...");
-    await this.changeRecords([
-      Record.forIdentity(this.domainName, identity.token!).action("DELETE"),
-    ]);
+    const existingTxtRecord = await this.getCurrentTXTRecord();
+    const existingTxtValues: Route53.ResourceRecord[] = existingTxtRecord?.ResourceRecords || [];
```

Comment:
Then this business logic can be expressed like this:

```ts
    const record = await this.getCurrentTXTRecord();
    record?.remove(identity.token);

    const recordSize = record?.size ?? 0;

    // If the record contained only the validation value, delete the record.
    // Otherwise just update the record with the value removed.
    if (recordSize === 0) {
      console.log("Deleting DNS Records used for domain verification...");
      await this.changeRecords([
        record.action("DELETE"),
      ]);
    } else {
      console.log("Updating DNS Records to remove the value used for domain verification...");
      await this.changeRecords([
        record.action("UPSERT"),
      ]);
    }
```